designer/client/src/components/graph/GraphContext.tsx

```diff
@@ -1,14 +1,25 @@
 /* eslint-disable i18next/no-literal-string */
-import React, { createContext, PropsWithChildren, useContext } from "react";
+import React, { createContext, PropsWithChildren, useContext, useMemo } from "react";
 import { Graph } from "./Graph";
 
 type GraphContextType = () => Graph;
 
 const GraphContext = createContext<GraphContextType>(null);
 
-export const GraphProvider = ({ graph, children }: PropsWithChildren<{ graph: GraphContextType }>) => (
-    <GraphContext.Provider value={graph}>{children}</GraphContext.Provider>
-);
+export const GraphProvider = ({
+    graph: _graphGetter,
+    children,
+}: PropsWithChildren<{
+    graph: GraphContextType;
+}>) => {
+    const graph = _graphGetter();
+
+    // force new value
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const graphGetter = useMemo(() => () => _graphGetter(), [_graphGetter, graph]);
```

Comment:
If i understand correctly, every component rerender, the new value is assigned, so IMO this useMemo has no sense or i missed something?

Reply:
new value is provided only when `graph` changes not every time `GraphProvider` renders.

it's sth like this
https://codesandbox.io/s/admiring-curie-dkms53?file=/src/App.js